### ISSUE TITLE
Building MethodDefinition with an ArrowFunctionExpression crashes the printer

### ISSUE THREAD

**Issue by ASDFGerte** [benjamn/recast#707]
I am moderately new to recast. While it is a fantastic library, there appears to be no extended documentation, that would include rules and examples for all functionality. Therefore, i sometimes go by type annotation.

`recast.types.builders.methodDefinition` takes a `FunctionKind` as third parameter. While the "finished product" will contain a `FunctionExpression`, it's plausible to me, that it would just internally convert other related types. However, giving it an `ArrowFunctionExpression` creates an AST, which crashes the printer:

    const recast = require("recast");

    let ast = recast.parse("");

    ast.program.body.push(recast.types.builders.classDeclaration(
        recast.types.builders.identifier("myClass"),
        recast.types.builders.classBody(
            [recast.types.builders.methodDefinition(
                "method",
                recast.types.builders.identifier("myMethod"),
                recast.types.builders.arrowFunctionExpression(
                    [],
                    recast.types.builders.blockStatement([])
                )
            )]
        )
    ));

    // printing the invalid (?) ast causes the exception.
    console.log(recast.print(ast).code);

> TypeError: Cannot read property 'map' of undefined

Using recast from NPM in node, `"recast": "^0.18.10"`

I cannot determine, whether this is a bug (in recast or ast-types), or simply misuse on my side (probably a bit of both). Also, any additional documentation apart from the related github pages would be beneficial, or maybe a place to ask, when needed (discord, irc, ...). In my case, I simply manually built a `FunctionExpression` from the `ArrowFunctionExpression` i had as input, used that, and it worked fine.

**Comment by eventualbuddha** [comment]
I don't believe it is supposed to be able to convert automatically, so doing the conversion yourself is right. I agree this could be clearer.

**Comment by jedwards1211** [comment]
@ASDFGerte FWIW, it's possible to do your AST modifications with `@babel/traverse`, and print with Recast, if that would be useful for you
